File: task-03/python/subtask-4.py
'''Write a program that reads a number n from a file named
input.txt and generates a diamond pattern of asterisks (*) in another file
named output.txt.'''

def read_number_from_file(filename):
    with open(filename, 'r') as inputfile:
        num = inputfile.read()
    return int(num.strip())

def write_number_to_file(filename, content):
    with open(filename, 'w') as infile:
        infile.write(content)

def generate_pattern(n):
    pattern = []

    for i in range(n):
        line = ' ' * (n - i - 1) + '*' * (2 * i + 1)
        pattern.append(line)
        
    for i in range(n - 2, -1, -1):
        line = ' ' * (n - i - 1) + '*' * (2 * i + 1)
        pattern.append(line)
        
    return "\n".join(pattern)

def write_pattern_to_file(filename, pattern):
    with open(filename, 'w') as outfile:
        outfile.write(pattern)

inputcontent = input("Enter the number: ")
write_number_to_file('input2.txt', inputcontent)
n = read_number_from_file('input2.txt')
result = generate_pattern(n)
write_pattern_to_file('output2.txt', result)

File: task-03/ruby/subtask-4.rb
def read_number_from_file(filename)
  num = File.read(filename)
  return num.strip.to_i
end

def write_number_to_file(filename, content)
  File.write(filename, content)
end

def generate_pattern(n)
  pattern = []

  for i in 0...n
    line = ' ' * (n - i - 1) + '*' * (2 * i + 1)
    pattern << line
  end

  for i in (n - 2).downto(0)
    line = ' ' * (n - i - 1) + '*' * (2 * i + 1)
    pattern << line
  end

  return pattern.join("\n")
end

def write_pattern_to_file(filename, pattern)
  File.write(filename, pattern)
end

print "Enter the number: "
inputcontent = gets.chomp
write_number_to_file('input4.txt', inputcontent)
n = read_number_from_file('input4.txt')
result = generate_pattern(n)
write_pattern_to_file('output4.txt', result)
